src/json-data.spec.js

```diff
@@ -0,0 +1,208 @@
+/* node: true */
+
+const plugin = require('./json-data');
+const mock = require('mock-fs');
+
+describe('JSON Data Plugin', () => {
+  beforeAll(() => {
+    mock({
+      'src/stache/data': {
+        'config.json': '{ "productNameLong": "Stache 2" }',
+        'test file.json': '{ "test":"test" }'
+      },
+      'foo.html': '<stache></stache>',
+      'shared/json-data.service.ts': `import { Injectable } from '@angular/core';
```

Comment:
Pulling the entire file into the test is excessive. The plugin is literally only looking for a specific file name and replacing the first `'noop'` it finds. The following should be enough to check that the plugin is doing what it's supposed to do:
```
const content = `private jsonData: any = 'noop';`;
```